File: 198-house-robber/198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        prev = nums[0]
        cur = max(nums[0], nums[1])
        for n in nums[2:]:
            temp = max(cur, n+prev)
            prev = cur
            cur = temp
        return cur

File: 198-house-robber/198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int first = 0;
        int second = nums[0];
        for(int i=1; i<nums.length; i++){
            int third = Math.max(second, first+nums[i]);
            first = second;
            second = third;
        }
        return Math.max(first, second);
    }
}